src/librustc_metadata/rmeta/encoder.rs

```diff
@@ -1710,10 +1710,15 @@ pub(super) fn encode_metadata(tcx: TyCtxt<'_>) -> EncodedMetadata {
         || {
             // Prefetch some queries used by metadata encoding
             tcx.dep_graph.with_ignore(|| {
-                par_for_each_in(tcx.mir_keys(LOCAL_CRATE), |&def_id| {
-                    tcx.optimized_mir(def_id);
-                    tcx.promoted_mir(def_id);
-                });
+                join(
+                    || {
+                        par_for_each_in(tcx.mir_keys(LOCAL_CRATE), |&def_id| {
+                            tcx.optimized_mir(def_id);
+                            tcx.promoted_mir(def_id);
+                        })
+                    },
+                    || tcx.exported_symbols(LOCAL_CRATE),
```

Comment:
Ah, so was this why the previous commit moved exported symbols later?

Reply:
Yes. It's moved later to give more time for prefetching to happen.